pkg/core/resources/apis/meshidentity/api/v1alpha1/helpers.go

```diff
@@ -32,11 +54,7 @@ func Matched(
 
 	var matches []scoredMatch
 
-	for _, mi := range meshIdentities {
-		if mi.Spec.Selector == nil || mi.Spec.Selector.Dataplane == nil || !mi.Spec.Selector.Dataplane.Matches(labels) {
-			continue
-		}
-
+	for _, mi := range AllMatched(labels, meshIdentities) {
```

Comment:
I just noticed, previous implem didn't check `mi.Status` but now in `AllMatched` you're also checking 
```go
if mi.Status != nil && (mi.Status.IsInitialized() || mi.Status.IsPartiallyReady()) {
    matches = append(matches, mi)
}
```
so IIUC `BestMatched` now won't return MeshIdentity unless it's either initialized or partially ready